File: src/CDK/Stacks/Repos.cs
using Amazon.CDK;
using Amazon.CDK.AWS.ECR;

namespace CDK {
  public interface IReposExports {
    IRepository RepoDeployer { get; set; }
    IRepository RepoRouter { get; set; }
    IRepository RepoReleaseApp { get; set; }
  }

  public class Repos : Stack, IReposExports {
    public IRepository RepoDeployer { get; set; }
    public IRepository RepoRouter { get; set; }
    public IRepository RepoReleaseApp { get; set; }

    public Repos(Construct parent, string id, IStackProps props) : base(parent, id, props) {
      //
      // Create new repositories
      // Note: these are in a distinct stack because they have to be
      // created and push'd to before a Lambda function referencing
      // them can be created.
      //
      RepoDeployer = new Repository(this, "repoDeployer", new RepositoryProps {
        RepositoryName = "microapps-deployer",
      });
      RepoRouter = new Repository(this, "repoRouter", new RepositoryProps {
        RepositoryName = "microapps-router",
      });
      RepoReleaseApp = new Repository(this, "repoRelease", new RepositoryProps {
        RepositoryName = "microapps-release",
      });
    }
  }
}

File: src/cdk/lib/Repos.ts
import * as cdk from '@aws-cdk/core';
import * as ecr from '@aws-cdk/aws-ecr';

export interface IReposExports {
  RepoDeployer: ecr.Repository;
  RepoRouter: ecr.Repository;
}

export class Repos extends cdk.Stack implements IReposExports {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // The code that defines your stack goes here
    this.RepoDeployer = new ecr.Repository(this, 'repoDeployer', {
      repositoryName: 'microapps-deployer',
    });
    this.RepoRouter = new ecr.Repository(this, 'repoRouter', {
      repositoryName: 'microapps-router',
    });
  }
  RepoDeployer: ecr.Repository;
  RepoRouter: ecr.Repository;
}
